app.rb

```diff
@@ -0,0 +1,41 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+# Methods to use with the application
+class App
+  def initialize
+    @books = []
+    @persons = []
+  end
+
+  attr_reader :persons, :books
+
+  # Method to create a book
+  def create_book(title, author)
+    book = Book.new(title, author)
+    @books.push(book)
+  end
+
+  # Method to create a person (teacher or student)
+  def create_student(name, age, parent_permission)
+    student = Students.new(name, age, parent_permission)
+    @persons << student
+  end
+
+  def create_teacher(name, age, specialization)
+    teacher = Teacher.new(name, age, specialization)
+    @persons << teacher
+  end
```

Comment:
- [ ] Great effort working on the suggestion requested by the previous reviewer, but currently, you only work on creating one of the person, which is the student, but when I tried to create a teacher I was getting an error. I would kindly suggest you work on it and be sure to have the app working perfectly

Screenshot 👇 

![oop](https://github.com/maths29/OOP-school-library/assets/78634841/6d0a2ee5-ec90-4216-b536-80d98cba57ef)